packages/trie/src/trie.ts

```diff
@@ -280,45 +280,49 @@ export class Trie {
   async findPath(key: Uint8Array, throwIfMissing = false): Promise<Path> {
     const stack: TrieNode[] = []
     const targetKey = bytesToNibbles(key)
+    const keyLen = targetKey.length
     let result: Path | null = null
-
+    let progress = 0
     const onFound: FoundNodeFunction = async (_, node, keyProgress, walkController) => {
-      // If we already have a result, exit early
-      if (result) return
-
-      if (node === null) {
-        result = { node: null, remaining: [], stack }
-        return
-      }
-
-      const keyRemainder = targetKey.slice(matchingNibbleLength(keyProgress, targetKey))
-      stack.push(node)
+      stack.push(node!)
```

Comment:
The only other thing I can come up with for potential performance improvement is to rework this bit here.  In theory, the stack could be created at the maximum possible length for the path of the node, so whatever the depth of the trie is (maybe 16)?  This `push` operation is still another destructive/constructive array operation that theoretically is more expensive than just storing something in the array since we're re-allocating memory here. So theoretically, you could pre-allocate the `stack` to be the maximum possible length, add items to it as you traverse the path of the key, and then return a slice of the path array equivalent to the length of the key, right?  That's the only other thing I can come up with unless @holgerd77 or @jochem-brouwer have other ideas.

Reply:
Are you saying that `stack.push(node)` is more expensive than if we start with  `stack = new Array(32)` and add via  `stack[i] = node` 

How would we know where to put a node in the stack array?

Somewhere along the way we will do:

`stack[ n ] = extension_node`

If the extension_node has nibble_length 3...

Would the next node in the stack belong at `stack[ n + 1 ]` or `stack[ n + 3 ]`?

To find `n + 1` we have to keep track of the index, and then slice the return stack.

`n + 3` would be equal to `progress` -- 
However we would have to iterate over the whole return stack to filter out `undefined`.,,

Reply:
I'm not 100% sure on the first point but push is having to allocate more memory every time you add to it and I think just copying a value is faster than allocating new memory and then copying it. Will need to do some simple tests to verify though. Let me think on it some more. We can always take this bit by bit